### PR TITLE
Closes #180 blog post unix vs sas datetime

### LINKED ISSUE #180: Blog Post: Unix versus SAS time origin
### Blog Post

In discussion in R Consortium Submission WG there was a finding around SAS versus Unix time.  Link and recording to come later

The CDISC Pilot Data update in Phuse Test Data Factory was using R unix time and gave +10 years to the dates. 😨 

The Pilot 3 re-derived ADaM data used xportr to help deal with this situation

image

A post should be written about this issue.  

More to come.

### PR DESCRIPTION
**Pull request by cpiraux**
Thank you for your Pull Request! We have developed this task checklist to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your blog post.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update), and make sure the corresponding issue is linked in the Development section on the right hand side
- [x] Run the script from `CICD.R` line by line to first check the spelling in your post and then to make sure your code is compatible with our code-style. Address any incongruences by following the instructions in the file! 
- [x] Choose (possibly several) `tag(s)` or categories from the current list: `c("Metadata", "SDTM", "ADaM", "TLG", "Shiny", "Community", "Conferences", "Submissions", "Technical")` for your blog post. If you cannot find anything that fits your blog post, add your own tag! We occasionally tidy up all `tags` for consistency.
- [x] Add a short description for your blog post in the `description` field at the top of the markdown document.
- [x] Blog post is *short*, *personalized*, *reproducible* and *readable*
- [ ] Add a disclaimer at the top of your post, if appropriate (example: **Disclaimer** 
This blog contains opinions that are of the authors alone and do not necessarily reflect the strategy of their respective organizations.)
- [x] Address all merge conflicts and resolve appropriately
- [ ] Assign two of us (@bms63, @manciniedoardo, @StefanThoma, @kaz462) as reviewers in the PR.
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by bms63 — COMMENTED** [review]
Could we do something with a clock or a countdown timer for the image 
image

File: posts/2024-07-08_unix_vs_sas_date.../unix_vs_sas_datetime.qmd [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+```{r setup, include=FALSE}
+long_slug <- "2024-07-08_unix_vs_sas_date..."
+# renv::use(lockfile = "renv.lock")
+```
+
+<!--------------- post begins here ----------------->
+
+Recent discussions within the R Consortium Submission Working Group have highlighted challenges in handling dates between SAS and Unix systems.
+In addition, during the CDISC Pilot Data update at the Phuse Test Data Factory, using R's Unix time format resulted in dates being mistakenly advanced by +10 years.
```
Comment (bms63) [inline posts/2024-07-08_unix_vs_sas_date.../unix_vs_sas_datetime.qmd:24]: Could we link the Phuse Test Data Factory - https://github.com/phuse-org/TestDataFactory.  Do you mind locating and verifying the time-stamp issue.  https://rconsortium.github.io/submissions-wg/minutes/2024-06-07/  Just want to make sure this is the right one!

Comment (bms63) [reply posts/2024-07-08_unix_vs_sas_date.../unix_vs_sas_datetime.qmd:24]: ACtually the meeting minutes might be nice to link as well!!

File: posts/2024-07-08_unix_vs_sas_date.../unix_vs_sas_datetime.qmd [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+Another important aspect to consider is the difference in data types for numeric dates in R and SAS.
+
+-   **SAS:** SAS only supports numeric and character data types.
+    Therefore, numeric dates in SAS need a format applied to be human-readable.
+
+-   **R:** In contrast, R has specific data types for dates, which allow numeric dates to be rendered with a date format directly in data frames.
+    The data types in R are: "Date" for dates, "hms" or "difftime" for time, and "POSIXct" or "POSIXt" for datetime.
+
+These differences can lead to discrepancies if not properly accounted for during data conversion, potentially resulting in a date shift of 10 years when transferring data between SAS and R.
```
Comment (bms63) [inline posts/2024-07-08_unix_vs_sas_date.../unix_vs_sas_datetime.qmd:50]: Could this be italicized?  It is so important!!

File: posts/2024-07-08_unix_vs_sas_date.../unix_vs_sas_datetime.qmd [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+## Considerations for Conversion
+
+To illustrate how date conversions between SAS and R can be managed, let's look at the [`haven`](https://haven.tidyverse.org/) package in R.
+This package facilitates the conversion of data between SAS and R.
+
+-   **From SAS to R:** Numeric variables with dates formats are identified as dates and converted to POSIX date(time) formats.
+
+    **Detection of date, time, datetime variable from SAS by haven package**
+
+    ![](haven_sas_to_R.png)
```
Comment (bms63) [inline posts/2024-07-08_unix_vs_sas_date.../unix_vs_sas_datetime.qmd:61]: Could you provide link to source code.  Is that C?

File: posts/2024-07-08_unix_vs_sas_date.../unix_vs_sas_datetime.qmd [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+[Dataset-JSON v1.1](https://wiki.cdisc.org/display/DSJSON1DOT1/Specification) could potentially be used in future data exchanges between SAS and R.
+Given the differing approaches to handling dates across programming languages, the challenge lies in managing dates within a programming language agnostic data exchange format.
+
+**What is Dataset-JSON?** Dataset-JSON is a dataset exchange standard that uses JSON.
+It is designed to meet regulatory submission requirements and other dataset exchange scenarios.
+It has the potential to replace XPT as the default format for clinical and device data submissions to regulatory authorities.
+
+To address the interoperability issue across programming languages, Dataset-JSON stores numeric date variables as strings in ISO 8601 format.
+Additionally, a `targetdatatype` attribute is included to differentiate between character and numeric dates within the dataset.
+This approach ensures that conversion tools receive sufficient information to handle dates correctly.
```
Comment (bms63) [inline posts/2024-07-08_unix_vs_sas_date.../unix_vs_sas_datetime.qmd:82]: Can we give a shout out to this R package as well! https://atorus-research.github.io/datasetjson/

**Review by bms63 — CHANGES_REQUESTED** [review]
Really awesome @cpiraux !!!

**Review by bms63 — COMMENTED** [review]
This image is amazing 
image

**Comment by bms63** [comment]
I think we just need to get rid of this message from dplyr (@StefanThoma I think this is from session update you did, any thoughts??) 
image

After that I think we are good to go!  @manciniedoardo any last thoughts?

**Comment by bms63** [comment]
oh and we need to fix the spelling workflow

**Comment by cpiraux** [comment]
Thanks, everyone, for your review!

I've implemented all the comments. I’ve just updated a paragraph on datasetJSON, as v1.1 is still a draft, by adding a reference to the specification for the latest updates. 

One of the checks is still failing. Do you know what to do to make this check pass?

**Comment by bms63** [comment]
umm..im not sure what is going on with the spelling check.  we can let it slide

**Pull request merged**